fibonacci/elixir/bench.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Bench do
+  def run do
+    {cli_num, ""} = Integer.parse(Enum.at(System.argv(), 0))
+
+    fib_res = sum_fibs(cli_num)
+
+    IO.puts(fib_res)
+  end
+
+  def fib(0), do: 0
+  def fib(1), do: 1
+  def fib(n), do: fib(n - 1) + fib(n - 2)
+
+  def sum_fibs(n), do: sum_range(0, 0, n)
+
+  defp sum_range(0, 0, n), do: sum_range(1, 0, n)
+
+  defp sum_range(i, acc, n) when i < n do
+    sum_range(i + 1, acc + fib(i), n)
+  end
+
+  defp sum_range(_, acc, _), do: acc
```

Comment:
According to some quick testing with `:benchee`, this version's about 0.118s faster on my machine.

```suggestion
  def run do
    cli_num = System.argv() |> Enum.at(0) |> String.to_integer()
    fib_res = sum_fibs(cli_num)
    IO.puts(fib_res)
  end

  defp sum_fibs(i \\ 0, max, total \\ 0)
  defp sum_fibs(i, max, total) when i < max, do: sum_fibs(i + 1, max, total + fib(i))
  defp sum_fibs(_i, _max, total), do: total

  defp fib(0), do: 0
  defp fib(1), do: 1
  defp fib(n), do: fib(n - 1) + fib(n - 2)
```

Reply:
Your version is way cleaner!
I haven't found any significant improvements on my machine though. It looks like it cuts one function call after all 🤔

Reply:
@MikaelFangel

It's fairly easy to do it iteratively instead of recursively, yes, but the benchmark is supposed to be done recursively. Take a look at some of the other languages' implementations, such as Go. Those are standard imperative languages with mutable state and yet they do it recursively, too.

Reply:
I know it’s supposed to be done recursively, hence my suggestion. Just to clarify, tail call optimizations (TCO) are just a specific form of recursion and not an iterative solution.

The other solutions mentioned can also be written in a uniform way that allows for TCO, which aligns with the rationale behind PR #55. This approach would better simulate real-world scenarios, where efficient recursion is often critical. A uniform TCO-enabled implementation across languages would make the benchmark both fairer and more reflective of practical usage.

Reply:
@MikaelFangel This is amazing! It makes the fibonacci down all the way from 1.8s to 400ms on my machine

I don't see why we shouldn't include it

Reply:
The solution you gave _is_ an iterative Fibonacci solution. The fact that it is written using recursive function calls is irrelevant. It is equivalent to doing

```go
a, b := 0, 1
for range n {
  a, b = b, a + b
}
return a
```

in Go. The entire point of the test is to cause the exponential recursive growth. If you do it the way that you're suggesting, it's not the same benchmark anymore.